spec/models/course/forum_spec.rb

```diff
@@ -24,5 +24,41 @@
         end
       end
     end
+
+    describe '.with_topic_count' do
+      let(:forum) { create(:forum) }
+      let!(:forum_topics) { create_list(:forum_topic, 2, forum: forum) }
+
+      it 'shows the correct count' do
+        expect(Course::Forum.with_topic_count.find(forum.id).topic_count).
+          to eq(forum_topics.size)
+      end
+    end
+
+    describe '.with_topic_post_count' do
+      let(:forum) { create(:forum) }
+      let(:first_topic) { create(:forum_topic, forum: forum) }
+      let(:second_topic) { create(:forum_topic, forum: forum) }
+      let!(:first_topic_posts) { create_list(:post, 2, topic: first_topic.acting_as) }
+      let!(:second_topic_posts) { create_list(:post, 1, topic: second_topic.acting_as) }
+
+      it 'shows the correct count' do
+        expect(Course::Forum.with_topic_post_count.find(forum.id).topic_post_count).
+          to eq(first_topic_posts.size + second_topic_posts.size)
+      end
+    end
+
+    describe '.with_topic_view_count' do
+      let(:forum) { create(:forum) }
+      let(:first_topic) { create(:forum_topic, forum: forum) }
+      let(:second_topic) { create(:forum_topic, forum: forum) }
+      let!(:first_topic_views) { create_list(:forum_topic_view, 2, topic: first_topic) }
+      let!(:second_topic_views) { create_list(:forum_topic_view, 1, topic: second_topic) }
+
+      it 'shows the correct count' do
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.